src/replication/status.cpp

```diff
@@ -42,13 +45,14 @@ void to_json(nlohmann::json &j, const ReplicationRoleEntry &p) {
   };
   auto processREPLICA = [&](ReplicaRole const &replica) {
     auto common = nlohmann::json{{kVersion, p.version},
-                                 {kReplicationRole, replication_coordination_glue::ReplicationRole::REPLICA},
-                                 {kPort, replica.config.port}};
+                                 {kReplicationRole, replication_coordination_glue::ReplicationRole::REPLICA}};
 
-    if (p.version == DurabilityVersion::V4) {
-      common[kAddress] = replica.config.address;
+    if (p.version == DurabilityVersion::V1 || p.version == DurabilityVersion::V2 ||
+        p.version == DurabilityVersion::V3) {
+      common[kIpAddress] = replica.config.repl_server.GetAddress();  // non-resolved
+      common[kPort] = replica.config.repl_server.GetPort();
     } else {
-      common[kIpAddress] = replica.config.address;
+      common[kReplicaServer] = replica.config.repl_server;  // non-resolved
```

Comment:
Again, I think this type of logic is a mistake. The to_json should only care about the current version.

Reply:
I agree about this one